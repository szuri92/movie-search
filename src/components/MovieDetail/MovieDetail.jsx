import React from 'react'
import { useSelector } from 'react-redux'
import { useParams, useHistory } from 'react-router-dom'

import styles from './MovieDetails.module.scss'

import ErrorBoundary from '../ErrorBoundary'
import Image from '../Image'

const MovieDetail = () => {
    const { id }  = useParams()
    const history = useHistory()

    const movie = useSelector(state => state.movies.movieList.filter(e => e.id == id))[0]

    //TODO: request movie by id if it's not yet in the store

    const handleGoBack = () => {
        history.goBack()
    }

    return (
        <ErrorBoundary>
            <section className={styles.detail_page}>
                <div>
                    <span className={styles.goBack} onClick={handleGoBack}> â—„ Back</span>
                    <h1>{ movie.title }</h1>
                    <h2>{ movie.release_date } </h2>
                    <h2>{ movie.popularity }</h2>
                    <div className={styles.overview}>{ movie.overview }</div>
                </div>
                <Image src={movie.poster_path}/>
            </section>
        </ErrorBoundary>
    )
}

export default MovieDetail