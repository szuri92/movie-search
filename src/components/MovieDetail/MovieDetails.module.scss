.detail_page {
    padding: 30px;
    display: flex;
    justify-content: space-around;

    .overview {
        max-width: 40vw;
    }
}