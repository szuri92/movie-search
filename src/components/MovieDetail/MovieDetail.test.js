import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import MovieDetail from './MovieDetail'

const mockFn = jest.fn()
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useParams: () => ({
      id: '1',
    }),
    useHistory: () => ({
        goBack: mockFn,
    })
}))

describe('MovieDetails', () => {

    const spy = jest.spyOn(redux, 'useSelector')
    spy.mockReturnValue([{
        id: 1,
        overview: 'overview',
        poster_path: 'poster_path',
    }])

    it('renders correctly', () => {
        const tree = shallow(<MovieDetail/>) 
        expect(shallowToJson(tree)).toMatchSnapshot()
    })

    it('renders correctly', () => {
        const tree = shallow(<MovieDetail/>) 
        const button = tree.find('span')
        button.simulate('click')
        expect(mockFn).toHaveBeenCalled()
    })
})

