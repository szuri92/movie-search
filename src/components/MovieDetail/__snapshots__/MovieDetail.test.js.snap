// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails renders correctly 1`] = `
<ErrorBoundary>
  <section
    className="detail_page"
  >
    <div>
      <span
        className="goBack"
        onClick={[Function]}
      >
         â—„ Back
      </span>
      <h1 />
      <h2>
         
      </h2>
      <h2 />
      <div
        className="overview"
      >
        overview
      </div>
    </div>
    <Image
      src="poster_path"
    />
  </section>
</ErrorBoundary>
`;
